packages/next/src/shared/lib/invariant-error.ts

```diff
@@ -0,0 +1,11 @@
+// Internal wrapper around build errors at development
+// time, to prevent us from propagating or logging them
+export class InvariantError extends Error {
+  constructor(message: string, options?: ErrorOptions) {
+    super(
+      `Invariant: ${message.endsWith('.') ? message : message + '.'} This is a bug in Next.js.`,
```

Comment:
The word "Invariant" can be confusing for users what the actually the error is. I suggest we simply remove it.

Does it only thrown at development or it also in prod build? If so should we get the flag not throw them in prod?

Reply:
> Does it only thrown at development or it also in prod build?
Well, depends on the feature, but in my case, they're meant to be thrown in prod as well.

This is really meant for internal assertions, and if you search around in the codebase, we prefix a lot of those with "Invariant: " as well, so i thought i'd be consistent with that convention. And really, the user isn't expected to understand this message, because it's something about Next internals -- that's what the "This is a bug in Next.js" is for.

Reply:
Then you need to change the comment `// Internal wrapper around build errors at development time`.

We need to be careful for adding errors, we had so many cases that assume it's a safe case to do assertion/invariant before that ended up with users hitting them in the corner cases. And we don't have any tests cases can cover them. Those errors caused some weird issues that we cannot patch back after released for a long while.

Reply:
omg, i accidentally left comment that comment in when copy pasting... 🤦‍♀️ 

> We need to be careful for adding errors

sure, but all of this is feature-flagged to be experiemental only.